includes/Core/Authentication/Setup_V2.php

```diff
@@ -54,15 +54,23 @@ public function handle_action_setup_start() {
 			wp_die( esc_html__( 'Site Kit is not configured to use the authentication proxy.', 'google-site-kit' ) );
 		}
 
-		if ( $this->credentials->has() ) {
-			$this->google_proxy->sync_site_fields( $this->credentials, 'sync' );
+		$required_scopes = $this->authentication->get_oauth_client()->get_required_scopes();
+		$this->google_proxy->with_scopes( $required_scopes );
+
+		$oauth_setup_redirect = $this->credentials->has()
+			? $this->google_proxy->sync_site_fields( $this->credentials, 'sync' )
+			: $this->google_proxy->register_site( 'sync' );
+
+		if ( is_wp_error( $oauth_setup_redirect ) || ! filter_var( $oauth_setup_redirect, FILTER_VALIDATE_URL ) ) {
+			wp_die( esc_html__( 'The request to the authentication proxy has failed. Please, try again later.', 'google-site-kit' ) );
 		}
 
 		if ( $redirect_url ) {
 			$this->user_options->set( OAuth_Client::OPTION_REDIRECT_URL, $redirect_url );
 		}
 
-		$this->redirect_to_proxy();
+		wp_redirect( $oauth_setup_redirect ); // phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect
+		exit;
 	}
 
```

Comment:
There is another problem further down in this file that I discovered during testing, but it's more of an oversight from #4376 than it belongs to this issue. Still, let's fix it in here: In `Setup_V2::handle_action_exchange_site_code`, if the request is successful, we currently redirect back to the proxy without adding the `site_id` query parameter which causes an exception.